### ISSUE TITLE
Error in importXenium with reading cell_boundaries.csv.gz with label_id column

### ISSUE THREAD

**Issue by saito-k1** [BIMSBbioinfo/VoltRon#112]
Hello,

I am attempting to load in our new Xenium dataset using the importXenium function with the sample_name specified, but I get this error: 

"Creating cell level assay ...
Warning: sparse->dense coercion: allocating vector of size 1.3 GiBmorphology_lowres.tif already exists!
Error in `as_tibble()`:
! Column 4 must be named.
Use `.name_repair` to specify repair.
Caused by error in `repaired_names()`:
! Names can't be empty.
✖ Empty name found at location 4.
Backtrace:
  1. VoltRon::importXenium(...)
  4. dplyr:::group_split.data.frame(., cell_id)
  6. dplyr:::group_by.data.frame(.tbl, ...)
  7. dplyr::grouped_df(groups$data, groups$group_names, .drop)
  8. dplyr:::compute_groups(data, vars, drop = drop)
 10. tibble:::as_tibble.data.frame(data)
 11. tibble:::lst_to_tibble(unclass(x), .rows, .name_repair)
 12. tibble:::set_repaired_names(...)
 13. tibble:::repaired_names(...)"

Any ideas on what the problem could be?  Thank you for your help!

**Comment by Artur-man** [comment]
Hello there, 

What I understand from the backtrace is that the problem is caused by the line in importXenium where cell segments are separated by cell_id column after the data frame read from cell_boundaries.csv.gz. 

This is how we do it: 

```
  # segments
  segments_file <- paste0(dir.path, "/cell_boundaries.csv.gz")
  if(file.exists(segments_file)){
    segments <- as.data.frame(data.table::fread(segments_file))
    colnames(segments) <- c("cell_id", "x", "y")
    if(use_image){
      segments[,c("x","y")] <- segments[,c("x","y")]/scaleparam
    }
    segments[,"y"] <- range_coords[2] - segments[,"y"]  + range_coords[1]
    segments <- segments %>% dplyr::group_split(cell_id)
    segments <- as.list(segments)
    names(segments) <- rownames(coords)
  } else {
    stop("There are no file named 'cell_boundaries.csv.gz' in the path")
  }
```

Could you check if you can read cell boundaries like above yourself and see if it works? Have you also tried this using the [dev branch](https://github.com/BIMSBbioinfo/VoltRon/tree/dev) ? 

Also, could you also share your full script, thanks so much!

**Comment by saito-k1** [comment]
Good morning! I tried using the dev branch with the same error. 

My full code is just loading the library and calling LoadXenium on my Xenium outs path: 
"library(VoltRon)
Xen_R1 <- importXenium("/Volumes/Xenium_Outs", sample_name = "XeniumR1")". 

I ran the code you provided and it errors at "segments[,c("x","y")] <- segments[,c("x","y")]/scaleparam". When I take a look at the cell_boundaries.csv.gz file, it looks like 10X has added a new column called label_id in the new version of their outs, which is what is causing the error. I can adjust this on my end for now, thank you!
